### PR TITLE
feat(49323): Render JSDoc @throws {type} as a link

### LINKED ISSUE #49323: Render JSDoc @throws {type} as a link
When you reference a type in a JSDoc `@throws`, it doesn't turn a passed `{type}` into a link` although we do turn `{@link Type}` to a link. It would be nice if such type references in JSDoc commands are also turned into a link.

image

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #49323

### CODE REVIEW

**Review by sandersn — CHANGES_REQUESTED** [review]

File: src/compiler/types.ts [hunk 1]
```diff
@@ -3573,6 +3574,11 @@ namespace ts {
         readonly typeExpression: JSDocSignature;
     }
 
+    export interface JSDocThrowsTag extends JSDocTag {
+        readonly kind: SyntaxKind.JSDocThrowsTag;
+        readonly name?: JSDocNameReference;
```
[the remark this answers is not in the document]
Comment (a-tarasyuk) [reply src/compiler/types.ts]: @sandersn Oke, I've changed to `typeExpression`.

File: tests/baselines/reference/quickInfoThrowsTag.baseline [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        {
+          "text": "void",
+          "kind": "keyword"
+        }
+      ],
+      "documentation": [],
+      "tags": [
+        {
+          "name": "throws",
+          "text": [
```
Comment (sandersn) [inline tests/baselines/reference/quickInfoThrowsTag.baseline:108]: shouldn't this include 'Error' somewhere? Maybe I'm misunderstanding the baseline json format, but I expected that the tag info should include the type name somewhere.

**Review by sandersn — CHANGES_REQUESTED** [review]
Looks good except one question about the quick info display.

File: tests/baselines/reference/quickInfoThrowsTag.baseline (hunk 2 above)
Comment (a-tarasyuk) [reply tests/baselines/reference/quickInfoThrowsTag.baseline:114]: I've added changes to include information about the type. @sandersn I noticed that the `@returns` tag also doesn't include this info. Is it ok? Or do we need to create a separate issue and fit it?
 
```
////class E extends Error {}
/////**
//// * @returns {E} comment
//// */
////function f1() {}
////f1/**/()

   "documentation": [],
   "tags":[
      {
         "name":"returns",
         "text":[
            {
               "text":"comment",
               "kind":"text"
            }
         ]
      }
   ]

```

**Review by sandersn — APPROVED** [review]

**Pull request merged**